src/core/xds/xds_client/xds_client.cc

```diff
@@ -1056,6 +1061,91 @@ void XdsClient::XdsChannel::AdsCall::ParseResource(
                                                 context->read_delay_handle);
 }
 
+void XdsClient::XdsChannel::AdsCall::HandleServerReportedResourceError(
+    size_t idx, absl::string_view resource_name, absl::Status status,
+    DecodeContext* context) {
+  std::string error_prefix = absl::StrCat(
+      "resource_errors index ", idx, ": ",
+      resource_name.empty() ? "" : absl::StrCat(resource_name, ": "));
+  if (resource_name.empty()) {
+    context->errors.emplace_back(
+        absl::StrCat(error_prefix, "resource_name unset"));
+    ++context->num_invalid_resources;
+    return;
+  }
```

Comment:
I think this would be easier to follow, no need to check the same condition twice:

```suggestion
    if (resource_name.empty()) {
      context->errors.emplace_back(
          absl::StrCat("resource_errors index ", idx, ": resource_name unset"));
      ++context->num_invalid_resources;
      return;
    }
    std::string error_prefix = absl::StrCat(
      "resource_errors index ", idx, ": ", resource_name, ": ");
```

Reply:
The way I have this now is the same pattern used in `ParseResource()` above, so I think it makes sense to stick with this for consistency.